src/as_execution/mod.rs

```diff
@@ -15,81 +15,178 @@ use crate::settings::max_number_of_pages;
 use crate::tunable_memory::LimitingTunables;
 use crate::GasCosts;
 
-pub(crate) use common::*;
 pub(crate) use context::*;
 pub(crate) use error::*;
 
 #[derive(Clone)]
 pub enum RuntimeModule {
-    ASModule((ASModule, Engine)),
+    ASModule(ASModule),
 }
 
 impl RuntimeModule {
     /// TODO: Dispatch module creation corresponding to the first bytecode byte
     ///
-    /// * (1) target AssemblyScript
+    /// * (1) TODO: target AssemblyScript (do not use ident)
     /// * (2) TODO: target X
-    /// * (_) target AssemblyScript and use the full bytecode
-    pub fn new(bytecode: &[u8], limit: u64, gas_costs: GasCosts) -> Result<Self> {
+    /// * (_) target AssemblyScript
+    pub fn new(
+        bytecode: &[u8],
+        limit: u64,
+        gas_costs: GasCosts,
+        cache_compatible: bool,
+    ) -> Result<Self> {
         match bytecode.first() {
-            Some(1) => Ok(Self::ASModule(ASModule::new(bytecode, limit, gas_costs)?)),
-            Some(_) => Ok(Self::ASModule(ASModule::new(bytecode, limit, gas_costs)?)),
+            Some(_) => Ok(Self::ASModule(ASModule::new(
+                bytecode,
+                limit,
+                gas_costs,
+                cache_compatible,
+            )?)),
             None => Err(anyhow!("Empty bytecode")),
         }
     }
+
+    pub fn serialize(&self) -> Result<Vec<u8>> {
```

Comment:
It seems to me that we are loosing some information during serialization (gas_costs, limit, cache_compatibible).

How can one deserialize here after?